StRoot/StEvent/StRHICfHit.h

```diff
@@ -0,0 +1,86 @@
+		void setMaxPeakBin(Int_t tower, Int_t layer, Int_t xy, Int_t val);
+
+		void setSingleHitNum(Int_t tower, Int_t layer, Int_t xy, Int_t val);
+		void setSingleHitPos(Int_t tower, Int_t layer, Int_t xy, Float_t val);
+		void setSinglePeakHeight(Int_t tower, Int_t layer, Int_t xy, Float_t val);
+		void setSingleFitChi2(Int_t tower, Int_t layer, Int_t xy, Float_t val);
+
+		void setMultiHitNum(Int_t tower, Int_t val);
+		void setMultiHitPos(Int_t tower, Int_t layer, Int_t xy, Int_t order, Float_t val);
+		void setMultiPeakHeight(Int_t tower, Int_t layer, Int_t xy, Int_t order, Float_t val);
+		void setMultiPeakRaw(Int_t tower, Int_t layer, Int_t xy, Int_t order, Float_t val);
+		void setMultiEnergySum(Int_t tower, Int_t layer, Int_t xy, Int_t order, Float_t val);
+		void setMultiFitChi2(Int_t tower, Int_t layer, Int_t xy, Float_t val);
+
+        Float_t getPlateEnergy(Int_t tower, Int_t plate);
+		Float_t getGSOBarEnergy(Int_t tower, Int_t layer, Int_t xy, Int_t bar);
+
+		Float_t getL20(Int_t tower);
+		Float_t getL90(Int_t tower);
+
+		Int_t getPointNum(Int_t tower);
+		Int_t getGSOMaxLayer(Int_t tower, Int_t order);
+		Int_t getMaxPeakBin(Int_t tower, Int_t layer, Int_t xy);
+
+		Int_t getSingleHitNum(Int_t tower, Int_t layer, Int_t xy);
+		Float_t getSingleHitPos(Int_t tower, Int_t layer, Int_t xy);
+		Float_t getSinglePeakHeight(Int_t tower, Int_t layer, Int_t xy);
+		Float_t getSingleFitChi2(Int_t tower, Int_t layer, Int_t xy);
+
+		Int_t getMultiHitNum(Int_t tower);
+		Float_t getMultiHitPos(Int_t tower, Int_t layer, Int_t xy, Int_t order);
+		Float_t getMultiPeakHeight(Int_t tower, Int_t layer, Int_t xy, Int_t order);
+		Float_t getMultiPeakRaw(Int_t tower, Int_t layer, Int_t xy, Int_t order);
+		Float_t getMultiEnergySum(Int_t tower, Int_t layer, Int_t xy, Int_t order);
+		Float_t getMultiFitChi2(Int_t tower, Int_t layer, Int_t xy);
+
+	private:
+		Float_t mPlateE[kRHICfNtower][kRHICfNplate];
+		Float_t mGSOBarSmallE[kRHICfNlayer][kRHICfNxy][kRHICfNbarSmall];
+		Float_t mGSOBarLargeE[kRHICfNlayer][kRHICfNxy][kRHICfNbarLarge];
+
+		Float_t mL20[kRHICfNtower];
+		Float_t mL90[kRHICfNtower];
+
+		Int_t mPointNum[kRHICfNtower];
+		Int_t mGSOMaxLayer[kRHICfNtower][kRHICfNorder];
+		Int_t mMaxPeakBin[kRHICfNtower][kRHICfNlayer][kRHICfNxy];
+
+		Int_t mSingleHitNum[kRHICfNtower][kRHICfNlayer][kRHICfNxy];
+		Float_t mSingleHitPos[kRHICfNtower][kRHICfNlayer][kRHICfNxy];
+		Float_t mSinglePeakHeight[kRHICfNtower][kRHICfNlayer][kRHICfNxy];
+		Float_t mSingleChiSquare[kRHICfNtower][kRHICfNlayer][kRHICfNxy];
+
+		Int_t mMultiHitNum[kRHICfNtower];
+		Float_t mMultiHitPos[kRHICfNtower][kRHICfNlayer][kRHICfNxy][kRHICfNorder];
+		Float_t mMultiPeakHeight[kRHICfNtower][kRHICfNlayer][kRHICfNxy][kRHICfNorder];
+		Float_t mMultiPeakRaw[kRHICfNtower][kRHICfNlayer][kRHICfNxy][kRHICfNorder];
+		Float_t mMultiEnergySum[kRHICfNtower][kRHICfNlayer][kRHICfNxy][kRHICfNorder];
+		Float_t mMultiChiSquare[kRHICfNtower][kRHICfNlayer][kRHICfNxy];
```

Comment:
That is an addition of 18 arrays some of which are 3 and 4 dimensional. Have you checked by how much the size of event.root files can increase on average?  These are static arrays so, if I think about this right all production chains will be affected, not just RHICf streams (if there is such thing) On the other hand, if they are empty the automatic ROOT compression should minimize the effect

Reply:
Hello Dmitri,
Thank you for your reply, and I understand about this delayed PR Thank you.

We including the RHICf collaborator and I already had a discussion about these variable arrays.
Actually, In the reconstruction process of the RHICf library, 
If some events not satisfying a condition (for example, there is no worth for reconstruction or no have a tower hit),
then We don't store these arrays (except mPlateE, mGSOBarSmallE, mGSOBarLargeE, mL20, and mL90)
But our collaborators told me these arrays, all of the various arrays might be needed.

In spite of that, I know what do you concern about.
As I tell you above, these arrays only store a few events per 100 events. 
(But if there is a satisfying saving condition, all of the arrays are stored fully.)
So I think I have to change these array formats from static array to vector.

Please let me know if these changes are inappropriate.

Thank you,
Seunghwan